components/data/data-services/org.wso2.micro.integrator.dataservices.core/src/main/java/org/wso2/micro/integrator/dataservices/core/DataServiceFault.java

```diff
@@ -197,7 +197,8 @@ public String getFullMessage() {
 			buff.append("Current Request Name: " + this.getCurrentRequestName() + "\n");
             getPropertyMap().put(DBConstants.FaultParams.CURRENT_REQUEST_NAME, this.getCurrentRequestName());
 		}
-		if (this.getCurrentParams() != null) {
+		String isCurrentParamsDisabled = System.getProperty(DBConstants.DISABLE_CURRENT_PARAMS_IN_LOG);
```

Comment:
Is it possible to read the property at server startup and not for every fault? It might affect the performance. WDYT?

Reply:
+1.